lib/cartopy/mpl/geoaxes.py

```diff
@@ -585,7 +585,18 @@ def set_extent(self, extents, crs=None):
 
         if projected is None:
             projected = self.projection.project_geometry(domain_in_crs, crs)
-        x1, y1, x2, y2 = projected.bounds
+        try:
+            # This might fail with an unhelpful error message ('need more
+            # than 0 values to unpack') if the specified extents fall outside
+            # the projection extents, so try and give a better error message.
+            x1, y1, x2, y2 = projected.bounds
```

Comment:
`projected.bounds` is just an empty tuple if this problem occurs. I guess a check would be fine too, but I can't think of another `ValueError` that'd come from just that line off the top of my head as it is just an assignment to unpack a tuple. Did you have something else in mind?

I thought an exception might actually be better than checking for empty, since it would always be triggered if the unpacking can't be done, I guess that corresponds to a check for `len(projected.bounds) == 4`...